theseus/optimizer/nonlinear/__init__.py

```diff
@@ -13,3 +13,4 @@
     NonlinearOptimizerParams,
     NonlinearOptimizerStatus,
 )
+from .trust_region import TrustRegionOptimizer
```

Comment:
We could drop `Optimizer` from name to `TrustRegion` for consistency with the other names.